### PR TITLE
MRG: refactoring of ECAT for clarity

### PR DESCRIPTION
**Pull request by matthew-brett**
Remove EcatMlist class for clarity.

Specify mlist as int32, data type from CTI sources.

Rewrite docstrings with better understanding of code.

### CODE REVIEW

**Comment by matthew-brett** [comment]
Any comments here?

**Comment by matthew-brett** [comment]
Comments here? I plan to merge soon, would very much like some comments / feedback.

**Comment by Eric89GXL** [comment]
I don't know anything about this data type, but I could give comments on any small things I find / make sure there are no potential glaring errors (although the CIs should tell us that to some extent). Would that be helpful?

**Comment by matthew-brett** [comment]
That would be excellent thanks - I'm no expert on ECAT either so any scan
for errors or bad style would be a big help.​

File: nibabel/ecat.py [hunk 1]
```diff
@@ -296,56 +330,57 @@ def read_mlist(fileobj, endianness):
...
+        # First row special, points to next mlist entries if present
+        n_unused, mlist_block_no, _, n_rows = rows[0]
+        if not (n_unused + n_rows) == 31: # Some error condition here?
             mlist = []
             return mlist
+        # Use all but first housekeeping row
         mlists.append(rows[1:n_rows+1])
         mlist_index += n_rows
-        if mlist_record_no <= 2: # should record_no in (1, 2) be an error?
+        if mlist_block_no <= 2: # should block_no in (1, 2) be an error?
```
Comment (Eric89GXL) [inline nibabel/ecat.py]: hah, found one! PEP8 two spaces before inline comment

Comment (Eric89GXL) [inline nibabel/ecat.py]: ...and elsewhere in the file if you want to follow it

File: nibabel/ecat.py [hunk 2]
```diff
@@ -441,67 +476,32 @@ def read_subheaders(fileobj, mlist, endianness):
...
-        mlist_data = read_mlist(fileobj, hdr.endianness)
-        self._mlist[:len(mlist_data)] = mlist_data
-
-    def get_frame_order(self):
-        return get_frame_order(self._mlist)
-
-    def get_series_framenumbers(self):
-        return get_series_framenumbers(self._mlist)
-
-
```
Comment (Eric89GXL) [inline nibabel/ecat.py]: nice, I like the simplification

**Comment by Eric89GXL** [comment]
Otherwise LGTM, but make sure the CI errors are make sense (esp. AppVeyor)

**Comment by matthew-brett** [comment]
Thanks for review - merging.

**Pull request merged**